### ISSUE TITLE
QueryManager should not forget active subscriptions

### ISSUE THREAD

**Issue by irminsul** [ember-graphql/ember-apollo-client#324]
Given a route `/foo` that has a query parameter `qp` configured with `refreshModel: true`, and a model hook that calls `this.apollo.subscribe(...)` (or `watchQuery()`), the following sequence of events will lead to a resource leak :

1. Visit `/foo?qp=1`.
2. The model hook executes and the query manager starts to track subscription A.
3. Change the value of `qp`.
4. [`beforeModel`](https://github.com/ember-graphql/ember-apollo-client/blob/master/addon/-private/apollo/setup-hooks.js#L9) is called and marks subscription A stale.
5. The model hook is called again and the query manager starts to track subscription B.
6. [`resetController`](https://github.com/ember-graphql/ember-apollo-client/blob/master/addon/-private/apollo/setup-hooks.js#L13) is called with `isExiting = false`.
7. [`unsubscribeAll`](https://github.com/ember-graphql/ember-apollo-client/blob/master/addon/-private/apollo/query-manager.js#L161) is called with `onlyStale = true`. The query manager unsubscribes subscription A (because it is stale), but it also forgets about subscription B because `activeSubscriptions` is unconditionally reset to an empty array:
    ```js
    unsubscribeAll(onlyStale = false) {
      this.activeSubscriptions.forEach(subscription => {
        if (!onlyStale || subscription.stale) {
          subscription.subscription.unsubscribe();
        }
      });
      this.activeSubscriptions = [];
    }
    ```
8. Subscription B now remains active indefinitely.

This problem is quite similar to the one described in #67, only that it results in a resource leak instead of a stuck route transition, and occurs even without the use of loading substates.

A possible fix cloud be to update `unsubscribeAll` so that it only stops tracking closed subscriptions:

```js
unsubscribeAll(onlyStale = false) {
  this.activeSubscriptions.forEach(subscription => {
    if (!onlyStale || subscription.stale) {
      subscription.subscription.unsubscribe();
    }
  });

  this.activeSubscriptions = this.activeSubscriptions.filter(subscription => {
    return !subscription.closed;
  });
}
```

**Comment by ChristianEllerbrock** [comment]
I ran into the same problems and can confirm the issue.

**Comment by josemarluedke** [comment]
This should be fixed in the last beta version v2.0.0-beta.8. Let me know if you find any issues.
